x-pack/plugins/triggers_actions_ui/public/application/sections/alert_details/components/alert_instances.tsx

```diff
@@ -134,18 +135,20 @@ export function AlertInstances({
   muteAlertInstance,
   unmuteAlertInstance,
   requestRefresh,
+  durationEpoch = Date.now(),
 }: AlertInstancesProps) {
   const [pagination, setPagination] = useState<Pagination>({
     index: 0,
     size: DEFAULT_SEARCH_PAGE_SIZE,
   });
 
+  const durationSinceEpoch = durationSince(durationEpoch);
```

Comment:
Actually the issue here isn't moment - it's that we're calculating the duration in the window as it's essentially a calculation of how long between *now* and when the instance was created.
In the test runner it's hard to know what *now* was when the component was rendered, so what I've done is store the *now* used by the render in a hidden field which we use to ensur it's been rendered correctly.

Before we were using a variable in the test itself that was set when the page is loaded - which obviously leaves room for flakiness.

I can't see a way around this as long as duration is calculated by the page load... once we have the event log this will be scrapped but for now I think this is the best approach.

Convinced? ;)

Reply:
Didn't realize the issue was we're display a time difference in the UI - eg, "this happened 2 minutes ago".  I can see how this can be problematic to test, without a fixed base time to operate on.  Thanks for explaining Gidi!